### PR TITLE
feat: interlink deployed versions, whether deprecated or not

### PR DESCRIPTION
**Pull request by CeerDecy**
#### What this PR does / why we need it:
interlink deployed versions, whether deprecated or not

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/all?id=615447&iterationID=-1&type=TASK)

#### Specified Reviewers:

/assign @iutx 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |      interlink deployed versions, whether deprecated or not        |
| 🇨🇳 中文    |       关联已经部署的版本，不管是否被废弃       |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by iutx — DISMISSED** [review]
LGTM
@sfwn  PTAL

**Review by iutx — DISMISSED** [review]
LGTM @sfwn

File: internal/tools/orchestrator/services/addon/addon.go [hunk 1]
```diff
@@ -3149,54 +3149,50 @@ func (a *Addon) deployAddons(req *apistructs.AddonCreateRequest, deploys []dbcli
...
 	if err != nil {
-		return "", "", err
+		return "", err
 	}
-	for _, r := range *registers {
-		dice, err := a.parseAddonDice(r)
-		if err != nil {
-			return "", "", err
+
+	if ins, err := a.db.FindTmcInstanceByNameAndCLuster(addonName, cluster); err == nil && ins != nil {
```
Comment (sfwn) [inline internal/tools/orchestrator/services/addon/addon.go:3162]: you should handle this error

**Review by sfwn — CHANGES_REQUESTED** [review]

**Review by sfwn — APPROVED** [review]

**Pull request merged**